subsys/random/Kconfig

```diff
@@ -17,6 +17,7 @@ config TEST_RANDOM_GENERATOR
 
 choice RNG_GENERATOR_CHOICE
 	prompt "Random generator"
+	default XOSHIRO_RANDOM_GENERATOR if RNG_GENERATOR_DEFAULT_XOSHIRO && ENTROPY_HAS_DRIVER
```

Comment:
> Kconfig does not allow to select an option which is a
> choice. Additional Kconfig option is added to allow that.

using a `default` for this probably won't work because a choice has a prompt, which means that if config in the choice is already active, then a selecting a symbol which then selects `RNG_GENERATOR_DEFAULT_XOSHIRO` which then changes the default choice will not have an effect because of the stuck symbol syndrome.

Ref: https://docs.zephyrproject.org/latest/build/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig

Reply:
what would be an alternative? Cannot do `select XOSHIRO_RANDOM_GENERATOR if ENTROPY_HAS_DRIVER` becuase it is a choice.

Note that this option is mainly for testing. In similar way we chose minimal logging for tests: https://github.com/zephyrproject-rtos/zephyr/blob/8c1efa8b96bb0209291f81840c115be4599c01ec/subsys/logging/Kconfig.mode#L7

Reply:
Do you need `&& ENTROY_HAS_DRIVER` condition there ? This symbols already depends on `ENTROPY_HAS_DRIVER`

Reply:
> Note that this option is mainly for testing. In similar way we chose minimal logging for tests:

and that suffers the same flaw (stuck symbol), I just tested it on a basic sample where I enabled logging.

Now `LOG_DEFAULT_MINIMAL` is a promptless symbol selected by `TEST_LOGGING_DEFAULTS`, which depends on `TEST`, so a user working with a regular sample would need to specify said sample as a test (CONFIG_TEST=y) to see this, and that is not likely to happen.

This means the main problem with having this solution in Kconfig code is mainly that others will see and copy the solution into other location, believing the code to work, and suddenly we have multiple places where `choices` suffers stuck symbol.

You said:
> what would be an alternative? 
> ....
> Note that this option is mainly for testing.

If this is for testing, then why the extra symbol indirection of `RNG_GENERATOR_DEFAULT_XOSHIRO` ?
As I see it, the `RNG_GENERATOR_DEFAULT_XOSHIRO` must already be selected by something else.
In this case test must have `CONFIG_ZTRESS=y`.

But why exactly is it not possible for tests that need a specific rng generator to have:
`CONFIG_XOSHIRO_RANDOM_GENERATOR=y` in their prj.conf, and be explicit about it ?

If there are good reasons why that is not a feasible approach, then maybe we should support a `Kconfig.defconfig` for tests, and load this early.
That would allow for a `subsys/testsuite/Kconfig.defconfig` where we could add:
```
# subsys/testsuite/Kconfig.defconfig

if TEST

choice RNG_GENERATOR_CHOICE
    default RNG_GENERATOR_DEFAULT_XOSHIRO if ZTRESS
endchoice

endif
```
that will be a regular change of defaults with well-known behavior for tests.

Not introducing a new symbol, which basically just increase complexity of Kconfig with no additional gain.

And we avoid misleading help texts like:
> Option can be used to force XOSHIRO generator instead of entropy device.

Reply:
@tejlmand I applied your suggestion. Please take another look.